### PR TITLE
Simple stdout/err console, via rudimentary IO override in python

### PR DESCRIPTION
**Pull request by sconover**
This is a good demonstration of what adding a new app module
entails.

### CODE REVIEW

File: src/ts/app/console/src/render.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import {html, TemplateResult} from "lit-html"
+import {ConsoleFragment} from "./subtree"
+
+
+
+export function render(fragments: ConsoleFragment[]): TemplateResult {
+    return html`
+        <style>
+        
+        div {
```
Comment (daicoden) [inline src/ts/app/console/src/render.ts:10]: I must not be understanding how the CSS scoping is working... this looks super dangerous to me, I would expect at least a class attached to the divs... but does this get scoped to only this component somehow? Who does that? lithtml?

File: src/ts/app/console/src/render.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        .stdout {
+            color: green
+        }
+
+        .stderr {
+            color: orange
+        }
+        </style>
+        
+        <div>${fragments.map((f) => html`<span class="${f.type}">${f.text}</span>`)}</div>
```
Comment (daicoden) [inline src/ts/app/console/src/render.ts:26]: eh, I don't think seeing stdout, and stderr is useful to kids, would rather highlight white or red.

File: src/ts/app/console/test/reducer-WRITE_TO_X-test.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import {assert} from "chai"
+import {suite, test} from "mocha"
+import {createReducer} from "../src/reducer"
+import {ConsoleFragmentType, Subtree} from "../src/subtree"
+import {ImprovedLoopForTesting} from "app/framework/test-support"
+import * as stateTransition from "../src/state-transition"
+import {programRunAction} from "app/program-run"
+
+suite("console reducer - WRITE_TO_X", () => {
```
Comment (daicoden) [inline src/ts/app/console/test/reducer-WRITE_TO_X-test.ts:9]: these tests are so cool

File: src/ts/app/framework/src/css-scope.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 export enum CssScope {
+    CONSOLE = "CONSOLE",
```
Comment (daicoden) [inline src/ts/app/framework/src/css-scope.ts:2]: ok... could you put a link to how this stuff works? I am toatally missing it.

File: src/ts/main/src/main.ts [hunk 5]
```diff
@@ -76,6 +77,34 @@ function main() {
         // for more about languagePluginLoader, see https://github.com/iodide-project/pyodide/blob/master/src/pyodide.js
         languagePluginLoader.then(() => {
 
+            //TODO: this needs to be a more complete python IO implementation,
+            // see https://github.com/python/typeshed/blob/5d553c9584eb86793cfa61019ddee5c7b62bc286/stdlib/2/typing.pyi#L320
+            (<any>window).fakeStdout = {
```
Comment (daicoden) [inline src/ts/main/src/main.ts:82]: nit: it's not fake, appStdout, consoleStdout, browserStdout...

File: src/ts/main/src/main.ts [hunk 6]
```diff
@@ -76,6 +77,34 @@ function main() {
...
+                write(str: string) {
+                    store.dispatch({
+                        type: programRunAction.Keys.WRITE_TO_STDOUT,
+                        text: str
+                    })
+                }
+            } as any
+
+            (<any>window).fakeStderr = {
```
Comment (daicoden) [inline src/ts/main/src/main.ts:91]: ditto

File: src/ts/app/console/src/render.ts (hunk 1 above)
Comment (sconover) [reply src/ts/app/console/src/render.ts:10]: Shadow dom! check out the subscribers...or just do a cross-project case-insensitive search for "shadow".

File: src/ts/app/console/src/render.ts (hunk 2 above)
Comment (sconover) [reply src/ts/app/console/src/render.ts:26]: Maybe not but surely it should be an option. Even if it's not the default.

All that's pending better default output choices (e.g. turtle, processing...).

File: src/ts/app/framework/src/css-scope.ts (hunk 4 above)
Comment (sconover) [reply src/ts/app/framework/src/css-scope.ts:2]: We need to talk live - let's go through it next time. It's all to do w/ the shadow dom / css isolation. 

This [bottom section](https://lit-html.polymer-project.org/guide/styling-templates#rendering-in-shadow-dom) covers some of it.

File: src/ts/main/src/main.ts (hunk 5 above)
Comment (sconover) [reply src/ts/main/src/main.ts:82]: This code is not production quality - this really needs to be a validly-duck-typed python IO implementation.

Can I rename it in a separate branch?

Comment (daicoden) [reply src/ts/main/src/main.ts:82]: sure

**Pull request merged**